server/middlewares/index.js

```diff
@@ -0,0 +1,65 @@
+const Document = require('../models/documents');
+
+module.exports = {
+  hasAccess: ((accessLevel, userId, params) => {
+    return ((req, res, next) => {
+      if (accessLevel.indexOf(req.decoded.role) > -1) {
+        if (req.decoded.role === 'user') {
```

Comment:
Check by something dynamic. Check by the roleIds specified for that particular document.